lib/init.js

```diff
@@ -0,0 +1,88 @@
+'use babel';
+
+import { parse, exec } from 'atom-linter';
+import { dirname, extname } from 'path';
+
+const isWindows = (/^win/).test(process.platform);
+const pattern = '.+:(?<line>\\d+):(?<col>\\d+)-(?<colEnd>\\d+):' +
+  ' \\((?<type>[EW])\\d+\\) (?<message>.*)';
+
+const checkedAppend = function checkedAppend(parameters, opt, args) {
+  if (args.length > 0) {
+    parameters.push(opt);
+    return parameters.concat(args);
+  }
+
+  return parameters;
+};
+
+const makeParameters = function makeParameters(globals, ignore, std, file) {
+  let parameters = [
+    '-', '--no-color', '--codes',
+    '--ranges', `--filename=${file}`
+  ];
+
+  parameters = checkedAppend(parameters, '--globals', globals);
+  parameters = checkedAppend(parameters, '--ignore', ignore);
+
+  if (std.length > 0) {
+    parameters.push('--std');
+    parameters.push(std.join('+'));
+  }
+
+  return parameters;
+};
+
+const reportToMessage = function reportToMessage(report, file) {
+  const end = report.range[1];
+  end[1] += 1;
+
+  return {
+    location: {
+      file,
+      position: [report.range[0], end]
+    },
+    severity: (report.type === 'E') ? 'error' : 'warning',
+    excerpt: report.text
+  };
+};
+
+export default {
+  activate() {
+    require('atom-package-deps').install('linter-luacheck');
```

Comment:
Ah, wow, missed that in this first pass. Definitely! Anything to cut out the activation time.